src/kubernetes_api_objects/custom_resource.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: MIT
+use crate::kubernetes_api_objects::common::*;
+use crate::kubernetes_api_objects::object_meta::*;
+use crate::pervasive::prelude::*;
+
+verus! {
+
+// TODO: CustomResource should be a generic type
+#[verifier(external_body)]
+pub struct CustomResource {
+    // the content is specific to the controller
```

Comment:
Is there no content (e.g. specific spec fields) in the CR that we need right now?

Reply:
Not for this PR. After this PR we can start with putting fields such like `replica` into CR and prove the same number of pods eventually are set up.